cpp/src/io/parquet/page_data.cu

```diff
@@ -1690,7 +1721,14 @@ extern "C" __global__ void __launch_bounds__(block_size)
           switch (dtype) {
             case INT32: gpuOutputFast(s, src_pos, static_cast<uint32_t *>(dst)); break;
             case INT64: gpuOutputFast(s, src_pos, static_cast<uint2 *>(dst)); break;
-            default: gpuOutputDecimalAsFloat(s, src_pos, static_cast<double *>(dst), dtype); break;
+            default:
+              // we currently do not support reading byte arrays larger than DECIMAL64
+              if (dtype_len <= 8) {
+                gpuOutputFixedLenByteArrayAsInt64(s, src_pos, static_cast<int64_t *>(dst));
```

Comment:
Should we do the same thing fir 32-bit values so we can store them as compactly as possible?

Reply:
It is perfectly fine to say "yes, in a different PR" 😄

Reply:
I am not sure whether it is necessary. In terms of spark-rapids, it only supports `DECIMAL64` (`INT64`) in current.